src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -337,7 +337,10 @@ private function createTypeAlternatives(TypedReference $reference)
             return ' '.$message;
         }
 
-        if (isset($this->ambiguousServiceTypes[$type])) {
+        $servicesAndAliases = $this->container->getServiceIds();
+        if (!$this->container->has($type) && false !== $key = array_search(strtolower($type), array_map('strtolower', $servicesAndAliases))) {
+            return sprintf(' Do you mean "%s" ?', $servicesAndAliases[$key], $type);
```

Comment:
Space before "?" Should be removed. Also, what about using preg_grep?

Reply:
Happy without the space for sure. preg_grep would be too much and we'd have find about all the escapes :/